docs/build_node_rust.md

```diff
@@ -0,0 +1,758 @@
+---
+id: build_node_rust
+title: "Building a Node with LDK in Rust"
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+## Introduction
+
+This document covers everything you need to make a node using LDK in Rust.
+
+* [Setup](#setup) covers everything you need to do to set up LDK on startup.
+* [Running LDK](#running-ldk) covers everything you need to do while LDK is running to keep it operational.
+
+Note that LDK does not assume that safe shutdown is available, so there is no 
+shutdown checklist.
+
+## Setup
+### 1. Initialize the `FeeEstimator`
+
+**What it's used for:** estimating fees for on-chain transactions that LDK wants broadcasted.
+
+**Example:**
+```rust
+impl FeeEstimator for YourFeeEstimator {
+	fn get_est_sat_per_1000_weight(
+		&self, confirmation_target: ConfirmationTarget,
+	) -> u32 {
+		match confirmation_target {
+			ConfirmationTarget::Background => // fetch background feerate,
+			ConfirmationTarget::Normal => // fetch normal feerate (~6 blocks)
+			ConfirmationTarget::HighPriority => // fetch high priority feerate
+		}
+	}
+}
+
+let fee_estimator = YourFeeEstimator::new();
```

Comment:
Do we want to assume that `fn new()` is a thing? We could add a `struct YourFeeEstimator()` at the top and then just do `YourFeeEstimator()` so that its at least copy-able.